File: BestTimetoBuyandSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
       int left = 0;
       int right = 1;
       int max = 0;
       while(right < prices.length){
           if(prices[right] - prices[left] > max){
               max = prices[right] - prices[left];
               right++;
           }
           else if(prices[right] - prices[left] < 0 && left < right-1){
               left++;
           }
           else if(prices[right] - prices[left] < 0){
               left++;
               right++;
           }
           else {
               right++;
           }
       }
       return max;
    }
}

File: BestTimetoBuyandSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left = 0
        right = 1
        maxval = 0
        while(right < len(prices)):
            sub = prices[right] - prices[left]
            if(sub > 0):
                maxval = max(maxval, sub)
            else:
                left = right
            right += 1
        return maxval

        